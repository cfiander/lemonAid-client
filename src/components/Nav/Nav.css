@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Sans|Playfair+Display&display=swap');

.mainNav {
 display: flex;
 justify-content: space-between;
 align-items: center;
 margin-top: 0;
 margin-bottom: 0;
 font-family: 'IBM Plex Sans', sans-serif;
 -webkit-box-shadow: 0 2px 4px -1px rgba(0,0,0,0.25);
	-moz-box-shadow: 0 2px 4px -1px rgba(0,0,0,0.25);
  box-shadow: 0 2px 4px -1px rgba(0,0,0,0.25);
  position: fixed;
  width: 100%;
}
.navlinks {
  justify-content: space-between;
}

.navColor {
  background-color: #548540;
}

ul {
  padding: 0;
  list-style: none;
}


.logo {
  max-height: 26px;
  margin-left: auto;
  margin-right: auto;
}

.homeButton {
  padding-left: 5%;
}

.register {
  padding-right: 5%;
}

@media screen and (max-width: 375px) {
   li {
    /* max-height: 26px; */
    
  }
}
