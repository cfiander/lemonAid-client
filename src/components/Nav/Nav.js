import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import './Nav.css'
import RecipeContext from '../../contexts/RecipeContext'
import logo from '../../images/LemonGreen.jpg'


export default class Nav extends Component {
  static contextType = RecipeContext

  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.context.setLogInFalse()
  }

  renderLogoutLink() {
    return (
      <li className='logoutButton'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </li>
    )
  }

  renderLoginLink() {
    return (
      <>
      <li className='loginButton'>
        <Link
          to='/login'>
          Log In
        </Link>
      </li>
      <li className="register">
        <Link
          to='/register'>
          Register
        </Link>
      </li>
      </>
    )
  }

  render() {
    return <>
      <ul className={`mainNav one-edge-shadow ${this.props.path !== '/' ? `navColor` : ''}`}>
        <li className="homeButton">
            <Link to={`/`}>Home</Link>
        </li>
        <li>
        <a href='#search'>
          Recipes
        </a>
        </li>
        <li><img className="logo" alt="logo" src={`${logo}`}/></li>
        {TokenService.hasAuthToken() && this.context.loggedIn
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </ul>
    </>
  }
}
