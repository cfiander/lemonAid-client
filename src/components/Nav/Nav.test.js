import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';
import toJson from 'enzyme-to-json'

describe('Nav component', () => {
  it('renders a Nav', () => {
    const wrapper = shallow(<Nav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
});