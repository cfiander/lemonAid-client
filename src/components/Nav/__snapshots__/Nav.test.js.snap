// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders a Nav 1`] = `
<Fragment>
  <ul
    className="mainNav one-edge-shadow navColor"
  >
    <li
      className="homeButton"
    >
      <Link
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        to="/recipes"
      >
        Recipes
      </Link>
    </li>
    <li>
      <Link
        to="/about"
      >
        <img
          alt="logo"
          className="logo"
          src="LemonGreen.jpg"
        />
      </Link>
    </li>
    <li
      className="loginButton"
    >
      <Link
        to="/login"
      >
        Log In
      </Link>
    </li>
    <li
      className="register"
    >
      <Link
        to="/register"
      >
        Register
      </Link>
    </li>
  </ul>
</Fragment>
`;
